Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+            version = "[variables('_solutionVersion')]";
+        }
+    }
+}
+
+function Get-ArmResource($name, $type, $kind, $properties){
+    [hashtable]$apiVersion = @{
+        "Microsoft.SecurityInsights/dataConnectors" = "2022-12-01-preview";
+        "Microsoft.SecurityInsights/dataConnectorDefinitions" = "2022-09-01-preview";
+        "Microsoft.OperationalInsights/workspaces/tables" = "2021-03-01-privatepreview";
+        "Microsoft.Insights/dataCollectionRules" = "2021-09-01-preview";
+    }
+
+    return [PSCustomObject]@{
+        name       = $name;
+        apiVersion = $apiVersion[$type]
+        type       = $type;
+        location   = "[parameters('workspace-location')]";
+        kind       = $kind;
+        properties = $properties;
+    }
+}
+
+function addNewParameter($templateResourceObj, $parameterName, $isSecret = $false) {
+    $hasParameter = [bool]($templateResourceObj.parameters.PSobject.Properties.name -match "$parameterName")
+    if (!$hasParameter) {
+        $templateResourceObj.parameters | Add-Member -NotePropertyName "$parameterName" -NotePropertyValue ([PSCustomObject] @{
+            defaultValue = $isSecret ? "-NA-" : "Enter $parameterName value";
+            type         = $isSecret ? "securestring" : "string";
+            minLength    = 1;
+        })
+    }
+
+    return $templateResourceObj;
+}
+
+# THIS IS THE STARTUP FUNCTION FOR CCP RESOURCE CREATOR
+function createCCPConnectorResources($contentResourceDetails, $dataFileMetadata, $solutionFileMetadata, $dcFolderName, $ccpDict, $solutionBasePath, $solutionName, $ccpTables, $ccpTablesCounter) {
+    Write-Host "Inside of CCP Connector Code!"
+    $solutionId = $solutionFileMetadata.publisherId + "." + $solutionFileMetadata.offerId
+    $placeHolderPatternMatches = '\{{[a-zA-Z0-9]+\}}'
+
+    if (!$global:baseMainTemplate.variables.workspaceResourceId) {
+        $global:baseMainTemplate.variables | Add-Member -NotePropertyName "workspaceResourceId" -NotePropertyValue "[resourceId('microsoft.OperationalInsights/Workspaces', parameters('workspace'))]"
+    }
+
+    if (!$global:baseMainTemplate.variables._solutionName) {
+        $global:baseMainTemplate.variables | Add-Member -NotePropertyName "_solutionName" -NotePropertyValue $dataFileMetadata.Name
+    }
+
+    if (!$global:baseMainTemplate.variables._solutionVersion) { 
+        $global:baseMainTemplate.variables | Add-Member -NotePropertyName "_solutionVersion" -NotePropertyValue $dataFileMetadata.Version
+    }
+
+    if (!$global:baseMainTemplate.variables._solutionId) { 
+        $global:baseMainTemplate.variables | Add-Member -NotePropertyName "_solutionId" -NotePropertyValue "$solutionId"
+    }
+
+    if (!$global:baseMainTemplate.variables."dataConnectorVersion$($templateKindByCounter[1])") { 
+        $global:baseMainTemplate.variables | Add-Member -NotePropertyName "dataConnectorVersion$($templateKindByCounter[1])" -NotePropertyValue "1.0.0"
```

Comment:
In the case of update of data connector, everytime this version value will come as by default 1.0.0?

Reply:
updated code